### PR TITLE
Issue 21801 moving folders

### LINKED ISSUE #21801: Moving Folders
Folders should always have the knowable/consistant id.

If someone wants to "move" a folder, we should not update the folder record. Instead, we should create a new folder record that has the correct id generated and delete the old folder / identifier when done.

I believe we will also need to update the info in the permission/permission_reference table as well.

### PR DESCRIPTION
**Pull request by nollymar**
This PR is the last step after this initial change: https://github.com/dotCMS/core/pull/21724

As now folder's identifier is unique and depends on its path, when a folder is moved a new identifier must be generated.
Some logic in the internalCheckin was modified to guarantee the identifier is updated before saving a new piece of content after a move. It fixes some broken tests.

### CODE REVIEW

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 1]
```diff
@@ -5030,15 +4976,18 @@ private Contentlet internalCheckin(Contentlet contentlet,
      * @throws DotSecurityException
      * @throws DotDataException
      */
-    private Contentlet assignIdentifierIfAbsent(final Contentlet contentlet, final Contentlet contentletRaw,  final String existingIdentifier, final String existingInode, final String htmlPageURL)
+    private Contentlet assignIdentifierIfAbsent(final Contentlet contentlet, final Contentlet contentletRaw,
```
[the remark this answers is not in the document]
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:4979]: Perhaps something like handleNewContentOrUpdate that name is crapy but the sense of that method has totally changed

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 2]
```diff
@@ -5069,6 +5018,58 @@ private Contentlet assignIdentifierIfAbsent(final Contentlet contentlet, final C
...
+            if(UtilMethods.isSet(contentletRaw.getFolder()) && !contentletRaw.getFolder().equals(FolderAPI.SYSTEM_FOLDER)){
+                final Folder folder = APILocator.getFolderAPI().find(contentletRaw.getFolder(), systemUser, false);
+                Identifier folderIdent = APILocator.getIdentifierAPI().find(folder.getIdentifier());
+                identifier.setParentPath(folderIdent.getPath());
+            } else {
+                identifier.setParentPath("/");
+            }
+            identifier = APILocator.getIdentifierAPI().save(identifier);
+
+            changedURI = ! oldURI.equals(identifier.getURI());
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:5072]: intellij gives me a warning about this value not getting modified. Maybe this method a `CheckinContext` Object with the contentlet and other things that are supposed to get  modified

File: dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImpl.java [hunk 3]
```diff
@@ -511,22 +512,26 @@ private boolean moveFile ( Contentlet fileAssetCont, Folder parent, Host host, U
...
-                    RefreshMenus.deleteMenu( oldParent );
                 }
+
                 CacheLocator.getNavToolCache().removeNav(oldParent.getHostId(), oldParent.getInode());
 
                 CacheLocator.getIdentifierCache().removeFromCacheByVersionable( fileAssetCont );
-
-				this.systemEventsAPI.pushAsync(SystemEventType.MOVE_FILE_ASSET, new Payload(fileAssetCont, Visibility.EXCLUDE_OWNER,
-						new ExcludeOwnerVerifierBean(user.getUserId(), PermissionAPI.PERMISSION_READ, Visibility.PERMISSION)));
+				HibernateUtil.addCommitListener(
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/business/FileAssetAPIImpl.java:526]: Nice!

**Review by fabrizzio-dotCMS — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:4979]: how about prepareForSave ?

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java [hunk 4]
```diff
@@ -5069,6 +5018,58 @@ private Contentlet assignIdentifierIfAbsent(final Contentlet contentlet, final C
...
+            } else if ( contentlet.isHTMLPage() ) {
+                //For HTML Pages - The asset name maps to the page URL
+                identifier.setAssetName( htmlPageURL );
+            }
+            if(UtilMethods.isSet(contentletRaw.getFolder()) && !contentletRaw.getFolder().equals(FolderAPI.SYSTEM_FOLDER)){
+                final Folder folder = APILocator.getFolderAPI().find(contentletRaw.getFolder(), systemUser, false);
+                Identifier folderIdent = APILocator.getIdentifierAPI().find(folder.getIdentifier());
+                identifier.setParentPath(folderIdent.getPath());
+            } else {
+                identifier.setParentPath("/");
```
[the remark this answers is not in the document]
Comment (nollymar) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:5068]: Done

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java (hunk 2 above)
Comment (nollymar) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:5072]: Fixed

File: dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (nollymar) [reply dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java:4979]: Fixed

**Review by fabrizzio-dotCMS — APPROVED** [review]

**Pull request merged**